File: all-submissions/0253-meeting-rooms-ii/solution.java
class Solution {
    public int minMeetingRooms(int[][] intervals) {
        
        if(intervals.length == 0) return 0;
        Arrays.sort(intervals, (a,b) -> a[0] - b[0]);
        PriorityQueue<Integer> queue = new PriorityQueue<>();
        int currMeetingRooms = 0;
        for(int[] interval: intervals){
            if(!queue.isEmpty() && queue.peek() <= interval[0]){
                queue.poll();
                queue.add(interval[1]);
            } else{
                queue.add(interval[1]);
                currMeetingRooms++;
            }
                     
        }
        return currMeetingRooms;
    }
}

File: all-submissions/0253-meeting-rooms-ii/solution.py
class Solution:
    def minMeetingRooms(self, intervals: List[List[int]]) -> int:
        if not intervals:
            return 0
        intervals.sort(key = lambda x: x[0])
        minHeap = []
        heapq.heappush(minHeap, intervals[0][1])
        for s_time,e_time in intervals[1:]:
            if s_time >= minHeap[0]:
                heapq.heappop(minHeap)
            heapq.heappush(minHeap, e_time)
        return len(minHeap)
